provisioning/atat_provisioning_api.yaml

```diff
@@ -0,0 +1,524 @@
+      operationId: deletePortfolioDraft
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+        '404':
+          description: Portfiolio Draft with the given ID does not exist
+  '/portfolioDrafts/{portfolioDraftId}/portfolio':
+    get:
+      tags:
+        - portfolioDraft
+      description: Gets the Portfolio Step of the Portfolio Draft Wizard
+      operationId: getPortfolioStep
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation with empty response body
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PortfolioStep'
+        '404':
+          description: Portfolio step not found for this Portfolio Draft
+        '504':
+          description: The given Portfolio Draft does not exist
+    post:
+      tags:
+        - portfolioDraft
+      description: Submits the Portfolio Step of the Portfolio Draft Wizard
+      operationId: createPortfolioStep
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation with empty response body
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PortfolioStep'
+        '405':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidationError'
+      requestBody:
+        description: Input data from the Portfolio Step of the Portfolio Draft Wizard
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PortfolioStep'
+  '/portfolioDrafts/{portfolioDraftId}/funding':
+    get:
+      tags:
+        - portfolioDraft
+      description: Gets the Funding Step of the Portfolio Draft Wizard
+      operationId: getFundingStep
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FundingStep'
+        '404':
+          description: Funding step not found for this Portfolio Draft
+        '504':
```

Comment:
504 does not feel like an appropriate response for a client error.

Reply:
Ah yes of course. I was trying to be sneaky and add some context for when the first element in the URI path does not exist (vs when the terminal element does, i.e. `/funding`).

But we can probably just make it a 400 when the underlying draft does not exist.